EnableIPythonDisplay.swift

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Hooks IPython to the KernelCommunicator, so that it can send display
+/// messages to Jupyter.
+
+import Python
+
+// Workaround SR-7757.
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+import func Darwin.C.dlopen
+#else
+import func Glibc.dlopen
+#endif
+dlopen("libpython2.7.so", RTLD_NOW | RTLD_GLOBAL)
+
+enum IPythonDisplay {
+  static var socket: PythonObject = Python.None
+  static var shell: PythonObject = Python.None
+}
+
+func enableIPythonDisplay() {
+  let json = Python.import("json")
+
+  let swift_shell = Python.import("swift_shell")
+  let socketAndShell = swift_shell.create_shell(
+    username: JupyterKernel.communicator.jupyterSession.username,
+    session_id: JupyterKernel.communicator.jupyterSession.id,
+    key: JupyterKernel.communicator.jupyterSession.key)
+  IPythonDisplay.socket = socketAndShell[0]
+  IPythonDisplay.shell = socketAndShell[1]
+
+  func updateParentMessage(to parentMessage: ParentMessage) {
+    IPythonDisplay.shell.set_parent(json.loads(parentMessage.json))
+  }
+  JupyterKernel.communicator.handleParentMessage(updateParentMessage)
+
+  func consumeDisplayMessages() -> [JupyterDisplayMessage] {
+    func bytes(_ py: PythonObject) -> [CChar] {
+      // faster not-yet-introduced method
+      // return py.swiftBytes!
+
+      // slow placeholder implementation
+      return py.map { el in
+        return CChar(bitPattern: UInt8(Python.ord(el))!)
+      }
+    }
+
+    let displayMessages = IPythonDisplay.socket.messages.map {
+      JupyterDisplayMessage(parts: $0.map { bytes($0) })
+    }
+    IPythonDisplay.socket.messages = []
+    return displayMessages
+  }
+  JupyterKernel.communicator.afterSuccessfulExecution(
+    run: consumeDisplayMessages)
+}
+
+enableIPythonDisplay()
```

Comment:
Could we add a global `display()` function like the following?

```swift
func display(_ item: Any) {
    
    if let pythonObject = item as? PythonObject {
        let display = Python.import("IPython.display")
        display.display(pythonObject)
    }
    else {
        // TODO: Add support for `CustomPlaygroundDisplayConvertible`
        print(item)
    }
}
```

Reply:
I want to keep this PR small and simple, so I won't add this in this PR.

I'm not sure we should have a function like that:
1. I think that in python notebooks, calling `IPython.display.display` is usually pretty uncommon, so it's not worth it to have a helper function for it. Usually you call methods specific to whatever you are displaying, like `plt.show()` for matplotlib.
2. We haven't thought about how Swift display APIs should work yet, so we shouldn't define a `display` function now that might constrain how they will work.

Reply:
Yes, that seems reasonable. I am trying to "trick" `pandas` into thinking it is in a IPython Notebook session so calling `display()` would not be needed to preview a dataframe.